flink-runtime/src/main/java/org/apache/flink/runtime/entrypoint/component/DispatcherResourceManagerComponent.java

```diff
@@ -88,25 +88,7 @@
 	}
 
 	private void registerShutDownFuture() {
-		terminationFuture.whenComplete(
-			(aVoid, throwable) -> {
-				if (throwable != null) {
-					shutDownFuture.completeExceptionally(throwable);
-				} else {
-					shutDownFuture.complete(ApplicationStatus.SUCCEEDED);
-				}
-			});
-
-		dispatcherRunner
-			.getTerminationFuture()
-			.whenComplete(
-				(aVoid, throwable) -> {
-					if (throwable != null) {
-						shutDownFuture.completeExceptionally(throwable);
-					} else {
-						shutDownFuture.complete(ApplicationStatus.SUCCEEDED);
-					}
-				});
+		FutureUtils.forward(dispatcherRunner.getShutDownFuture(), shutDownFuture);
```

Comment:
Also forward `terminationFuture`?

Reply:
No, the termination of the `JobManagerRunner` should not trigger the shut down logic handling.

Reply:
I'm not quite sure what `terminationFuture` represents here...Isn't it the termination of `DispatcherResourceManagerComponent`? Where `JobManagerRunner` comes from?

Reply:
Sorry I meant the `DispatcherRunner`.

Reply:
Exactly Tison. And if the shut down future is not completed then because the owner of this component initiated its shutdown and hence already knows that the component is terminating.